lib/services/database.dart

```diff
@@ -10,18 +10,32 @@ class DatabaseService {
   final CollectionReference userCollection =
       FirebaseFirestore.instance.collection('students');
 
-  Future<void> updateStudent(String name, String yearId) async {
-    return await userCollection.doc(uid).set({
-      'name': name,
-      'yearId': yearId,
+  Future<void> setStudent({String name, String yearId}) async {
+    return await userCollection
+        .doc(uid)
+        .set({'name': name, 'yearId': yearId, 'favoriteTAs': []});
+  }
+
+  Future<void> updateStudent(
```

Comment:
Suggestion: use doc.set with merge:true instead of two different methods. [Link](https://stackoverflow.com/a/63834370/9405663).

Reply:
thanks, will check

Reply:
deleted